### PR TITLE
[pigeon] 0.2.0 Made null-safe generation the default, updated documentation 

### PR DESCRIPTION
**Pull request by gaaclarke**
Most of the work for 0.2.0 happened in an external contributors commit: https://github.com/flutter/packages/pull/300

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by gaaclarke** [comment]
cc @Goddchen

**Comment by Goddchen** [comment]
@gaaclarke I have the fixes for the tests ready. But I don't know how I can get them in here. I tried checking out this repo/branch, committed them locally and tried to push them. But obviously I don't have write permission in here...

**Comment by Goddchen** [comment]
Hopefully this will to the trick: https://github.com/gaaclarke/packages/pull/1

**Comment by gaaclarke** [comment]
@Goddchen Thanks, that was kind of you.  There is something weird going on with our CI.  They are running locally fine and I can't retry.  I'll keep your PR handy while I try to kick the bots.

**Comment by Goddchen** [comment]
What's going on with the checks? They seem to be hanging again?

**Comment by gaaclarke** [comment]
@Goddchen yea, I'm pinging a few people from my side to see if I can get it resolved.

**Comment by mitchross** [comment]
Seeing a crash running this.

` flutter pub run 
pigeon --input pigeons/messages.dart --dart_null_safety
Unhandled exception:
Null check operator used on a null value
#0      generateJava (package:pigeon/java_generator.dart:248:49)
#1      Pigeon.run.<anonymous closure> (package:pigeon/pigeon_lib.dart:486:34)
#2      Pigeon._runGenerator (package:pigeon/pigeon_lib.dart:357:9)
#3      Pigeon.run (package:pigeon/pigeon_lib.dart:484:15)
<asynchronous suspension>
#4      main (file:///C:/Users/xxxx/AppData/Local/Temp/flutter_pigeon.94186523/_pigeon_temp_.dart:8:17)    
<asynchronous suspension>`
image

**Comment by gaaclarke** [comment]
@mitchross Weird, it's working in the tests and when I try it out.  Any idea what might be different for you?
```
$ git status
On branch pigeon-2-0-0
Your branch is up to date with 'origin/pigeon-2-0-0'.

$ pub run pigeon --input pigeons/message.dart --dart_null_safety --java_out foo.java --dart_out /dev/null 
```

What does your messages.dart file look like?

**Comment by alr2413** [comment]
@gaaclarke,
The same happens for me.
You just need to clone the flutter vlc player repository [https://github.com/solid-software/flutter_vlc_player.git](https://github.com/solid-software/flutter_vlc_player.git)
and then inside the flutter_vlc_player folder run pigeon command.

**Comment by mitchross** [comment]
> @gaaclarke,
> The same happens for me.
> You just need to clone the flutter vlc player repository https://github.com/solid-software/flutter_vlc_player.git
> and then inside the flutter_vlc_player folder run pigeon command.

Also swap out the pubspec 

dev_dependencies:
  flutter_test:
    sdk: flutter
  pigeon: 
    git:
      url: https://github.com/gaaclarke/packages.git
      path: packages/pigeon
      ref: pigeon-2-0-0`

**Comment by gaaclarke** [comment]
@mitchross @alr2413 Thanks, should be fixed now.

File: packages/pigeon/README.md [hunk 1]
```diff
@@ -112,9 +114,9 @@ public interface Api2Host {
 
 ## Null Safety (NNBD)
```
Comment (xster) [inline packages/pigeon/README.md:115]: We should caveat this a bit more clearly. If we're highlighting null safety more prominently, we should be clear there are 2 layers. There's the general null safety concept and there we're not doing too much yet. Things can't be defined as non-null and the nullable types aren't specifically annotated as so in kotlin and swift. There's also the Dart 2.12 null safety mechanism. We don't support non-null neither but we properly tag everything as nullable now.

**Review by xster — APPROVED** [review]

**Comment by Goddchen** [comment]
`Failing after 1441m` These checks are really screwed sometimes... I wish there was a simple "retry" button...

File: packages/pigeon/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
+## 0.2.0
+
+* **BREAKING CHANGE** - Pigeon files must be null-safe now.  That means the
+  fields inside of the classes must be declared nullable (non-null fields arent't
+  yet supported).  Migration example:
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply packages/pigeon/CHANGELOG.md]: done

File: packages/pigeon/README.md (hunk 1 above)
Comment (gaaclarke) [reply packages/pigeon/README.md:115]: done

**Comment by gaaclarke** [comment]
Okay, published.  Thanks for the patience.

**Pull request merged**